src/build/env.py

```diff
@@ -141,15 +154,23 @@ def __enter__(self):  # type: () -> IsolatedEnvironment
         if 'PATH' in os.environ:
             exe_path.append(os.environ['PATH'])
 
-        self._replace_env('PATH', os.pathsep.join(exe_path))
-        self._replace_env('PYTHONPATH', os.pathsep.join(sys_path))
-        self._replace_env('PYTHONHOME', self.path)
-        self._pop_env('PIP_REQUIRE_VIRTUALENV')
-
         self._symlink_relative(sysconfig.get_path('include'))
         self._symlink_relative(sysconfig.get_path('platinclude'))
         self._symlink_relative(sysconfig.get_config_var('LIBPL'))
 
+        self._symlink_file(sys.executable, os.path.join('bin', 'python'))
+        exe_path.insert(0, os.path.join(self.path, 'bin'))
```

Comment:
this path worth saving a variable, as you should reuse it two lines below.

Reply:
I would have to use `os.path.join` anyway, so I didn't feel the need. I can change that if you still want.

```python
bin_path = os.path.join(self.path, 'bin')
exe_path.insert(0, bin_path)
sys.executable = os.path.join(bin_path, 'python')
```

Reply:
I personally prefer this over that .